### PR TITLE
Link Cubyz more correctly.

### LINKED ISSUE #460: Distributable linking
Cubyz is actually really poorly set up for building to distribute on mac. I suspect the same might be true for linux as well.

Currently, Cubyzig has the following rpaths:
```
     path /Users/USER/.cache/zig/p/12204ba799cc74baeec8284b8a22cd0b597b58d5fa32c6eac999635fdc1834c950fc/lib (offset 12)
     path /usr/local/GL/lib (offset 12)
     path /Users/USER/Cubyz/Library (offset 12)
     path /opt/homebrew/Cellar/libx11/1.8.9/lib (offset 12)
     path /Users/USER/.cache/zig/p/12204ba799cc74baeec8284b8a22cd0b597b58d5fa32c6eac999635fdc1834c950fc/lib (offset 12)
```

There are a number of problems:
- cubyz_deps is listed twice (it shouldn't be at all, since it's a static library).
- The Library path is nonsensical because Zig evaluated it at build-time instead of runtime.
- Cubyz should link against XQuartz X11, not homebrew.
- X11 doesn't need an RPATH.

So, at present, every single rpath is just plain invalid for distribution.

Ideally, the rpaths would just be the following:
```
    path @executable_path/../Library (if app bundle)
    path /usr/local/GL/lib (if local build)
```

But I'm not sure if this is even possible within Zig build! Zig seems to discourage linking libraries by their paths alone (demanding an entire compile step or just a query to pkgconfig), and forcefully includes some (incorrect) rpath.

### PR DESCRIPTION
**Pull request by archbirdplus**
This PR attempts to resolve #460.

### CODE REVIEW

**Comment by archbirdplus** [comment]
Notably, this links the static cubyz_deps as a static library rather than a runtime dependency.

**Comment by archbirdplus** [comment]
Also allows Cubyz to compile on macos once again, it's been broken for a while.

File: saves/Development/world.dat [hunk 1]
```diff
@@ -4,6 +4,7 @@
 		59786,
 		100
 	],
+	"biomeChecksum" : 6660618543759111321,
```
Comment (IntegratedQuantum) [inline saves/Development/world.dat:7]: Please revert this.

File: build.zig [hunk 2]
```diff
@@ -21,10 +21,10 @@ pub fn build(b: *std.Build) !void {
...
 	exe.linkLibCpp();
 
-	const depsLib = b.fmt("cubyz_deps_{s}-{s}-{s}", .{@tagName(t.cpu.arch), @tagName(t.os.tag), switch(t.os.tag) {
-		.linux => "musl",
-		.macos => "none",
-		.windows => "gnu",
+	const depsLib = b.fmt("libcubyz_deps_{s}-{s}-{s}", .{@tagName(t.cpu.arch), @tagName(t.os.tag), switch(t.os.tag) {
+		.linux => "musl.a",
+		.macos => "none.a",
+		.windows => "gnu.a",
```
Comment (IntegratedQuantum) [inline build.zig:27]: This doesn't work on windows, since the name and extension of the static lib are different.

Comment (archbirdplus) [reply build.zig:27]: I'll add it in another case.

**Comment by archbirdplus** [comment]
We really gotta get #453 done already.

**Pull request merged**